### PR TITLE
Vision quest part 0: Remove Channel from Connection

### PR DESCRIPTION
**Pull request by andrewgordstewart**
This is the first three commits of https://github.com/statechannels/go-nitro/pull/513:

- TwoPartyLedgerChannel is removed from Connection
- build errors are fixed
- some client code is changed:
  - `MyIndex` is a public field on `ConsensusChannel`: https://github.com/statechannels/go-nitro/pull/513/commits/d0e34c5759bd0dfb4f7ff8d4178fbe4a1a669c05
  - GetConsensusChannelById is added to the store (note that TwoPartyLedger can currently be fetched by Id)

### CODE REVIEW

File: protocols/virtualfund/virtualfund-single-hop_test.go [hunk 1]
```diff
@@ -2,12 +2,10 @@ package virtualfund
```
Comment (andrewgordstewart) [inline protocols/virtualfund/virtualfund-single-hop_test.go:2]: This file is slated for deletion, so you can ignore the diffs!

File: protocols/virtualfund/helpers_test.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
+package virtualfund
```
Comment (andrewgordstewart) [inline protocols/virtualfund/helpers_test.go:1]: This file is copy-past code from virtualfund-single-hop_test.go.

File: protocols/virtualfund/virtualfund.go [hunk 3]
```diff
@@ -38,17 +38,8 @@ type GuaranteeInfo struct {
...
-}
-
-// ledgerChannelAffordsExpectedGuarantees returns true if, for the channel inside the connection, and for each asset keying the input variables, the channel can afford the allocation given the funding.
-// The decision is made based on the latest supported state of the channel.
-//
-// Both arguments are maps keyed by the same asset.
-func (c *Connection) ledgerChannelAffordsExpectedGuarantees() bool {
-	return c.Channel.Affords(c.getExpectedGuarantees(), c.Channel.OnChainFunding) // todo: #420 deprecate
+	Channel       *consensus_channel.ConsensusChannel
+	GuaranteeInfo GuaranteeInfo
```
Comment (andrewgordstewart) [inline protocols/virtualfund/virtualfund.go:42]: The changes made to this file are more or less forced by removing the TwoPartyLedger from a Connection.

File: client/engine/store/mockstore.go [hunk 4]
```diff
@@ -206,6 +206,27 @@ func (ms *MockStore) GetTwoPartyLedger(firstParty types.Address, secondParty typ
 	return ledger, ok
 }
 
+// GetConsensusChannel returns a ConsensusChannel between the calling client and
+// the supplied counterparty, if such channel exists
```
Comment (lalexgap) [inline client/engine/store/mockstore.go:210]: This godoc needs to be updated for `GetConsensusChannelById`

**Review by lalexgap — APPROVED** [review]

File: client/engine/store/mockstore.go (hunk 4 above)
Comment (andrewgordstewart) [reply client/engine/store/mockstore.go:210]: 5f374aa95af8714380d83631430525a709793867 (in Part 1)

**Pull request merged**